core/src/Dscp/Core/Foundation/Witness.hs

```diff
@@ -173,14 +173,15 @@ instance Buildable TxWitnessed where
 
 -- | Transaction for private block publications.
 data PublicationTx = PublicationTx
-    { ptAuthor    :: Address
-    , ptPrevBlock :: Maybe PrivateHeaderHash
-    , ptBlock     :: PrivateHeaderHash
+    { ptAuthor     :: Address
+    , ptHeader     :: PrivateBlockHeader
+    , ptPrevHeader :: Maybe PrivateBlockHeader
```

Comment:
What? Why not hash in `ptPrevHeader`? We already have that entity in our world state, why duplicate?

Reply:
Oh yeah my god I'm kind of stupid. Give me a sec.